test/remote/gateways/remote_safe_charge_test.rb

```diff
@@ -52,9 +52,7 @@ def test_successful_regular_purchase_through_3ds_flow_with_non_enrolled_card
   def test_successful_regular_purchase_through_3ds_flow_with_invalid_pa_res
     response = @three_ds_gateway.purchase(@amount, @three_ds_invalid_pa_res_card, @three_ds_options)
     assert_success response
-    assert !response.params['acsurl'].blank?
-    assert !response.params['pareq'].blank?
-    assert !response.params['xid'].blank?
+    assert_equal 'Attempted But Card Not Enrolled', response.params['threedreason']
```

Comment:
This fix is for a failing remote test unrelated to the `verify` changes.

This test covers a case where a 3ds transaction returns a Payer Response (`pa_res`) indicating that the 3ds flow has been rejected.

Quoted from Nuvei/SafeCharge's Gateway Direct API Integration Guide:

> If the customer's card is not enrolled in 3D Secure, SafeCharge returns the regular Gateway parameters...and in addition the parameters ECI and ThreeDReason.

Based on this, the `acsurl`, `pareq`, and `xid` fields should not have a value for this particular response.

>